demo/ios/ForegroundApp/ForegroundApp/ViewModel.swift

```diff
@@ -0,0 +1,151 @@
+
+    let accessKey = "${YOUR_ACCESS_KEY_HERE}" // Obtained from Picovoice Console (https://console.picovoice.ai)
+    let language: String = ProcessInfo.processInfo.environment["LANGUAGE"] ?? "en"
+
+    private var porcupineManager: PorcupineManager?
+
+    func setupWakeWords() {
+        if language == "en" {
+            wakeWordKeys = Porcupine.BuiltInKeyword.allCases.map { $0.rawValue }
+        } else {
+            if let keywordUrls = Bundle.main.urls(forResourcesWithExtension: "ppn", subdirectory: "keywords") {
+                wakeWordKeys = keywordUrls.map {
+                    $0.lastPathComponent
+                        .replacingOccurrences(of: "_ios.ppn", with: "")
+                        .replacingOccurrences(of: "_", with: " ")
+                }
+            }
+        }
+        selectedWakeWord = wakeWordKeys.first ?? ""
+    }
+
+    func toggleListening() {
+        if isListening {
+            stopListening()
+        } else {
+            startListening()
+        }
+    }
+
+    func startListening() {
+        guard VoiceProcessor.hasRecordAudioPermission else {
+            VoiceProcessor.requestRecordAudioPermission { isGranted in
+                if isGranted {
+                    DispatchQueue.main.async {
+                        self.startListening()
+                    }
+                } else {
+                    DispatchQueue.main.async {
+                        self.showError("Microphone permission is required")
+                    }
+                }
+            }
+            return
+        }
+
+        let keywordCallback: ((Int32) -> Void) = { _ in
+            DispatchQueue.main.async {
+                self.backgroundColor = .orange
+                DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+                    self.backgroundColor = Color(UIColor.systemBackground)
+                }
+            }
+        }
+
+        let errorCallback: ((Error) -> Void) = { error in
+            DispatchQueue.main.async {
+                self.showError(error.localizedDescription)
+            }
+        }
+
+        do {
+            if language == "en" {
+                guard let keyword = Porcupine.BuiltInKeyword(rawValue: selectedWakeWord)
+                else {
+                    showError("Keyword not found")
+                    return
+                }
+                porcupineManager = try PorcupineManager(
+                    accessKey: accessKey,
+                    keyword: keyword,
+                    onDetection: keywordCallback,
+                    errorCallback: errorCallback
+                )
+            } else {
+                guard let keywordUrl = Bundle.main.url(
+                        forResource: "\(selectedWakeWord.lowercased())_ios",
+                        withExtension: "ppn",
+                        subdirectory: "keywords")
+                else {
+                    showError("Keyword not found")
+                    return
```

Comment:
this can be more helpful, e.g. Keyword file for ${keyword} not found at ${filepath}